### PR TITLE
Replace Travis CI with GitHub Actions workflows

### PR DESCRIPTION
**Pull request by schlessera**

### CODE REVIEW

File: .github/workflows/code-quality.yml [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+jobs:
+
+  lint: #-----------------------------------------------------------------------
+    name: Lint PHP files
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out source code
+        uses: actions/checkout@v2
+
+      - name: Set up PHP envirnoment
```
Comment (jrfnl) [inline .github/workflows/code-quality.yml:18]:
```suggestion
      - name: Set up PHP environment
```
(here and everywhere else)

**Comment by jrfnl** [comment]
Had a quick look through what is here so far (this is not a full review).

* You may want to have a look at the [`ramsey/composer-install`](https://github.com/ramsey/composer-install) action - this will save on a lot of duplication with the composer install/composer caching.
* IMO the `lint` action should be run - at least - on the lowest + highest supported PHP version.
    For a library like this which is used as a dependency for other projects, I'd even go so far as to run it on all supported PHP versions just to be sure.
* Pro-tip: for the `lint` action, exclude the `.git` directory.
    This is not excluded by default in Parallel Lint and for "old" projects, not excluding it can really slow the action down _a lot_ as the script will search the whole `.git` directory for PHP files.
    Oh and for this repo, you may also want to exclude the `docs` directory, though that shouldn't make much difference.
* Suggestion: add a `lint` script to the `composer.json` file to make it easy for people to run the linting locally with the correct parameters:
    ```json
    "lint": [
      "@php ./vendor/php-parallel-lint/php-parallel-lint/parallel-lint . -e php --exclude vendor --exclude .git
    ],
    ```
    Note:
    * the `@php` is to make sure the same PHP version is used as used by Composer (instead of a default PHP version on a system);
    * the full path name is for improved compatibility cross-platform (i.e. make sure it also works on Windows).
* I hadn't come across the `cs2pr` usage before, but if it does what I expect it to do, I **_like_** it 👍 
    Will have to nick that trick and apply it to various workflows in other projects 😉 
* I generally prefer to run the QA checks against all pushes, not just `master`, though for the tests - depending on how complex the matrix is - I sometimes use a simpler, only high/low job for pushes.
    Providing people check the action outcome before opening a PR, it should prevent PRs being opened which don't pass on the basics, which then have to get a second commit/force push to fix the PR as soon as it's opened.
* You can add PHP `8.1` to the matrix for the `unit` job to test against the current PHP `nightly`.
    Use it in combination with `continue-on-error` to prevent it from failing/stopping the workflow run.
* Once the script is finished, you may want to remove the `fail-fast: false` to be kind to the GH resources.

**Comment by jrfnl** [comment]
Just making a note here that this PR would close #420 + #439

**Comment by jrfnl** [comment]
Oh and another reminder: let's not forget to update the badges in the `Readme` once this PR is ready.

**Comment by jrfnl** [comment]
I'm moving this back to the 2.0.0 milestone as we do still seem to be able to view the Travis runs, which allows us to release 1.8.0 knowing that CI is passing, so we can leave this until after the 1.8.0 release.

**Comment by schlessera** [comment]
Closing this for now, as it is being superseded by #467

**Pull request closed without merging**